daml-script/dump/src/main/scala/com/daml/script/dump/TreeUtils.scala

```diff
@@ -50,9 +59,29 @@ object TreeUtils {
     }
   }
 
+  private def layerSelectors(events: Seq[TreeEvent.Kind]): Seq[Selector] = {
+    val created = mutable.HashMap.empty[TemplateId, Int].withDefaultValue(0)
+    val exercised = mutable.HashMap.empty[(TemplateId, Choice), Int].withDefaultValue(0)
+    events.collect {
```

Comment:
You might want to error out on `Kind.Empty` here instead of using `collect`. If you ever actually get this (which could happen if we expand the ledger API), the `zip` below will blow up since you have fewer selectors than root events.

Reply:
Good point. I've avoided the issue by pulling the iteration into `layerSelectors`, renaming it to `foreachEventWithSelector`. This way we can just skip `Kind.Empty` without having to throw an error or providing an empty selector of some sort.